secret.go

```diff
@@ -36,8 +37,8 @@ func applySecret(ctx *cli.Context) error {
 	orgIdentifier := ctx.String("org-id")
 	projectIdentifier := ctx.String("project-id")
 	requiresFile := isFileTypeSecret(secretType)
-	secretIdentifier := getSecretIdentifier(secretType)
-	secretName := getSecretName(secretType)
+	secretName := ctx.String("secret-name")
+	secretIdentifier := getSecretIdentifier(secretName)
```

Comment:
This is a different field and not same as secretName. Please check.

Reply:
In the UI the Harness entity ID is generated based off the entity name (minus non-alphanumeric characters). 

Right now the CLI auto-creates a secretName and secretID based off a switch case of the secretType. 

This change modifies so user specified secretName, and the getSecretIdentifier function generates the ID secretName. 

Example:
`./harness secret apply --token <value> --secret-name "docker-secret"

<img width="928" alt="Screenshot 2023-12-26 at 11 18 26 AM" src="https://github.com/harness/harness-cli/assets/142037330/61edf055-c2f0-4254-aac1-ad298b5201b5">

No other dependencies on the secret identifier far as I can tell.

Reply:
Unless we want to change so that the secretIdentifier is also set by the user as a command line flag (rather than generated from secretName)